src/gsym.rs

```diff
@@ -0,0 +1,90 @@
+use std::fs::File;
+use std::io::{Error, Read};
+use std::mem;
+use std::path::PathBuf;
+
+use super::{AddressLineInfo, FindAddrOpts, SymResolver, SymbolInfo};
+
+mod parser;
+mod types;
+
+use parser::{find_address, GsymContext};
+
+/// The symbol resolver for the GSYM format.
+pub struct GsymResolver {
+    file_name: PathBuf,
+    ctx: GsymContext<'static>,
+    #[allow(dead_code)]
+    data: Vec<u8>,
+    loaded_address: u64,
+}
+
+impl GsymResolver {
+    pub fn new(file_name: PathBuf, loaded_address: u64) -> Result<GsymResolver, Error> {
+        let mut fo = File::open(&file_name)?;
+        let mut data = vec![];
+        fo.read_to_end(&mut data)?;
+        let ctx = GsymContext::parse_header(&data)?;
+
+        Ok(GsymResolver {
+            file_name,
+            ctx: unsafe { mem::transmute(ctx) },
```

Comment:
In my opinion it should be documented why this `transmute` is safe. We are artificially extending the lifetime without any indication why that is kosher.